lib/sidekiq_base.cc

```diff
@@ -274,86 +280,127 @@ float sidekiq_base<HdlType>::read_temperature() {
 
 
 template<typename HdlType>
-bool sidekiq_base<HdlType>::start_streaming() {
-	bool result{true};
+int sidekiq_base<HdlType>::start_streaming() {
+    int status;
+    if (dual_channel) {
+        status = sidekiq_functions.start_streaming_func(card, hdl2);
+        if (status != 0) {
+            printf("Error: could not start streaming, status %d, %s\n", status, strerror(abs(status)) );
+            exit(status);
+        }
+    }
 
-	if (sidekiq_functions.start_streaming_func(card, hdl) != 0) {
-		printf("Error: could not start streaming\n");
-		result = false;
+	status = sidekiq_functions.start_streaming_func(card, hdl);
+	if (status != 0) {
+		printf("Error: could not start streaming, status %d, %s\n", status, strerror(abs(status)) );
+        exit(status);
 	}
-	return result;
+	return status;
 }
 
 template<typename HdlType>
-bool sidekiq_base<HdlType>::stop_streaming() {
-	bool result{true};
+int sidekiq_base<HdlType>::stop_streaming() {
+    int status;
+    if (dual_channel) {
+        status = sidekiq_functions.stop_streaming_func(card, hdl2);
+        if (status != 0) {
+            printf("Error: could not stop streaming, status %d, %s\n", status, strerror(abs(status)) );
+        }
+    }
 
-	if (sidekiq_functions.stop_streaming_func(card, hdl) != 0) {
-		printf("Error: could not stop streaming\n");
-		result = false;
+	status = sidekiq_functions.stop_streaming_func(card, hdl);
+	if (status != 0) {
+		printf("Error: could not stop streaming, status %d, %s\n", status, strerror(abs(status)) );
 	}
-	return result;
+	return status;
 }
 
 template<typename HdlType>
-double sidekiq_base<HdlType>::get_sample_rate() {
+double sidekiq_base<HdlType>::get_sample_rate(HdlType handle) {
+    int status;
 	uint32_t p_rate;
 	double p_actual_rate;
 	uint32_t p_bandwidth;
 	uint32_t p_actual_bandwidth;
 
-	if (sidekiq_functions.get_sample_rate_func(card, hdl, &p_rate, &p_actual_rate, &p_bandwidth, &p_actual_bandwidth) !=
-		0) {
-		printf("Error: failed to set sample rate or bandwidth\n");
+	status = sidekiq_functions.get_sample_rate_func(card, handle, &p_rate, &p_actual_rate, &p_bandwidth, &p_actual_bandwidth);
+	if (status != 0) {
+		printf("Error: failed to get sample rate, status %d, %s\n", status, strerror(abs(status)) );
+        exit(status);
```

Comment:
do you really want this exit call here?  In general, I would think you just want to propagate the error and let the upper layers decide if it's fatal or not.

Reply:
Yeah but to do that we would have to return something like a 0 for rate or something.  I was under the philosophy of if there are any skiq errors we should bail since something bad is happening.

Thoughts?

Reply:
I'd suggest throwing an runtime exception.  I believe there's precedence in GNU radio for throwing exceptions and this is the perfect case to use that rather than exit.  In general, error handling should be left up to the user IMHO.

Reply:
ok will look around for a "gnuradio bail out exception" thanks.